lib/NetworkManager.js

```diff
@@ -20,12 +20,12 @@ const Multimap = require('./Multimap');
 class NetworkManager extends EventEmitter {
   /**
    * @param {!Puppeteer.CDPSession} client
-   * @param {!Puppeteer.FrameManager} frameManager
+   * @param {!Puppeteer.Page} page
    */
-  constructor(client, frameManager) {
+  constructor(client, page) {
```

Comment:
I don’t think you need to pass the page into the network manager anymore

Reply:
With the refactoring that I've done, we do.
Both FrameManager and NetworkManager are just wrappers for some Page functionality. So instead of passing them into each other and resorting to ugly hacks to ensure none of them is `undefined`, I'm simply passing a reference to the Page that generated them both.
And yes, FrameManager is still used here in [`_handleRequestStart`](https://github.com/GoogleChrome/puppeteer/blob/bbd1543ebd6148112d1e0d133530053775587b1d/lib/NetworkManager.js#L244).